src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -118,12 +123,30 @@ public HttpConnectionPool(HttpConnectionPoolManager poolManager, HttpConnectionK
                     break;
             }
 
+            if (_http2Enabled && _poolManager.Settings._maxHttpVersion != HttpVersion.Version20)
+            {
+                _http2Enabled = false;
+            }
```

Comment:
Instead of:
```C#
_http2Enabled = true;
```
earlier, should we instead just assign it there and remove this if?
```C#
_http2Enabled = _poolManager.Settings._maxHttpVersion >= HttpVersion.Version20;
```

Reply:
There are two cases where we set _http2Enabled to true, so I wanted to do the maxHttpVersion check in one place.

Alternatively, we could take the opposite approach and initialize _http2Enabled to true if maxHttpVersion allows it, then do the switch and set it to false in the cases where HTTP2 isn't allowed.  I'm happy either way.